lib/Mojo/IOLoop.pm

```diff
@@ -158,6 +164,12 @@ sub subprocess {
 
 sub timer { shift->_timer(timer => @_) }
 
+sub transition {
+  my ($self, $id, $class) = (_instance(shift), @_);
+  my $new = $class->new($self->stream($id)->steal_handle);
+  return $self->_stream($new, $self->_id, !!$self->{in}{$id});
```

Comment:
How is the old stream garbage collected? Can't we just reuse the same id? Once the handle is stolen the old stream is dead anyway.